File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

input = sys.stdin.readline

n = int(input())
m = int(input())

connected = [[] for _ in range(n + 1)]
infected = [False] * (n + 1)
stack = []
result = 0

for _ in range(m):
    a, b = map(int, input().split())
    connected[a].append(b)
    connected[b].append(a)

infected[1] = True
for elem in connected[1]:
    stack.append(elem)

while stack:
    idx = stack.pop()
    if not infected[idx]:
        infected[idx] = True
        result += 1
        for elem in connected[idx]:
            if not infected[elem]:
                stack.append(elem)

print(result)

File: 백준/Silver/2606. 바이러스/바이러스.swift
let n = Int(readLine()!)!
let e = Int(readLine()!)!
var graph = [Int: [Int]]()
var result = 0
var visited = [Bool](repeating: false, count: n + 1)

for _ in 0..<e {
    let input = readLine()!.split(separator: " ").compactMap { Int($0) }
    graph[input[0], default: []].append(input[1])
    graph[input[1], default: []].append(input[0])
}
dfs(1)
print(result)


func dfs(_ x: Int) {
    visited[x] = true
    for next in graph[x, default: []] {
        if !visited[next] { 
            result += 1
            dfs(next) 
        }
    }
}